src/qibolab/platforms/native.py

```diff
@@ -0,0 +1,188 @@
+from collections import defaultdict
+from dataclasses import dataclass, field, fields
+from typing import Dict, List, Optional, Union
+
+from qibo.config import raise_error
+
+from qibolab.pulses import FluxPulse, PulseConstructor, PulseSequence, PulseType
+
+
+@dataclass
+class NativePulse:
+    """Container with parameters required to generate a pulse implementing a native gate."""
+
+    name: str
+    duration: int
+    amplitude: float
+    shape: str
+    type: str
+    qubit: "Qubit"
+    frequency: int = 0
+    relative_start: int = 0
+    """Relative start is relevant for two-qubit gate operations which correspond to a pulse sequence."""
+
+    def __post_init__(self):
+        if self.type != "ro" and self.type != "qd":
+            raise_error(TypeError, f"NativePulse was initialized with incorrect type {self.type}.")
+
+    @classmethod
+    def from_dict(cls, name, **kwargs):
+        """Parse the dictionary provided by the runcard.
+
+        Args:
+            name (str): Name of the native gate (dictionary key).
+            kwargs (dict): Dictionary containing the parameters of the pulse
+                implementing the gate.
+        """
+        field_names = {field.name for field in fields(cls)}
+        kwargs = {k: v for k, v in kwargs.items() if k in field_names}
+        return cls(name, **kwargs)
+
+    def pulse(self, start, relative_phase=0.0):
+        """Construct the :class:`qibolab.pulses.Pulse` object implementing the gate.
+
+        Args:
+            start (int): Start time of the pulse in the sequence.
+            relative_phase (float): Relative phase of the pulse.
+
+        Returns:
+            A :class:`qibolab.pulses.DrivePulse` or :class:`qibolab.pulses.DrivePulse`
+            or :class:`qibolab.pulses.FluxPulse` with the pulse parameters of the gate.
+        """
+        pulse_type = PulseType(self.type).name
+        pulse_cls = PulseConstructor[pulse_type].value
+        channel = getattr(self.qubit, pulse_type.lower()).name
+        return pulse_cls(
+            start + self.relative_start,
+            self.duration,
+            self.amplitude,
+            self.frequency,
+            relative_phase,
+            self.shape,
+            channel,
+            qubit=self.qubit.name,
+        )
+
+
+@dataclass
+class NativeFluxPulse(NativePulse):
+    """Container with parameters required to create a :class:`qibolab.pulses.FluxPulse` object."""
+
+    def __post_init__(self):
+        if self.type != "qf":
+            raise_error(TypeError, f"NativeFluxPulse was initialized with incorrect type {self.type}.")
```

Comment:
In the spirit of the enum above, you could simply overwrite the `type` class attribute, attributing as type a new enum extending the original one.